iwyu_tool_test.py

```diff
@@ -97,6 +97,11 @@ def test_from_compile_command(self):
             [iwyu_tool.IWYU_EXECUTABLE, '-foo', '-Iinclude', 'file.cc'])
         self.assertEqual(invocation.cwd, '/home/user/llvm/build')
 
+    def test_compdb_dir(self):
+        compilation_db = iwyu_tool.parse_compilation_db('tests/tool/compdb-dir')
```

Comment:
I'm not a fan of test data files. It would be nice if we could make `parse_compilation_db` accept a file-object or a path. Or, since the function appears to do two thing (reading from file and canonicalizing), maybe move the file reading into `main` and add a `def fixup_compilation_db(compilation_db)` that does both this directory fixup and the realpath fixup. I chose `fixup` over `canonicalize_...` because the latter is too long :)

Reply:
I'm a little short on time, but let me know if you want me to do the refactor and push to master so you can rebase onto something that's simpler to test.

Reply:
If you could do that, that would be great. :-) Failing that, I probably will have time to do this the next weekend.

Reply:
I had something like this in mind: https://github.com/include-what-you-use/include-what-you-use/pull/718

Reply:
Thanks. I've subscribed to that, so I can rebase this on top of a new master once it's in.

Reply:
Should be in place now, see if you can retrofit this on top of the new testcase.

Reply:
Rebasing on top of master was easy, let me look at the testing side now.